### DISCUSSION TITLE
"Forbidden" error when trying to unfollow some user

### DISCUSSION THREAD

**Discussion by loicdescotte** [plhery/node-twitter-api-v2#196] [Q&A] [answered]
Hi,

I am trying to write scripts to cleanup a twitter timeline, in my first tests I have issues to use the API to unfollow a user : 

```js
const {TwitterApi} = require('twitter-api-v2');

const client = new TwitterApi({
    appKey: 'xxx',
    appSecret: 'xxx',
    accessToken: 'xxx',
    accessSecret: 'xxx'
  });

const idToUnfollow = 'xxx'

client.v2.me().then((me) => {
    //this is working : 
    console.log("me", me)
    
    //but not this : 
    client.v2.unfollow(me.data.id, idToUnfollow)
    .catch((error) => console.log(error)) // <- error here
  } 
)
```
I'm using some real values instead of 'xxx' instead of course :)

`console.log("me", me)`  is working, it is displaying information about my user.
But `client.v2.unfollow(me.data.id, idToUnfollow)` is not working . I have this error : 

```
 error: true,
  type: 'response',
  code: 403,
  headers:
   { 'content-type': 'application/problem+json',
     'cache-control': 'no-cache, no-store, max-age=0',
     'content-length': '93',
     'x-response-time': '102',
     'x-connection-hash':
      '2ef1f74face1dc57515760c0a9c89b0ae10f674d5e80630f511f665fb4f1de1e',
     date: 'Sun, 20 Feb 2022 13:01:38 GMT',
     server: 'tsa_f',
     connection: 'close' },
  rateLimit: undefined,
  data:
   '{\n  "title": "Forbidden",\n  "type": "about:blank",\n  "status": 403,\n  "detail": "Forbidden"\n}' }
```

Did I miss something?
Thanks :)

**Comment by alkihis** [comment, marked as answer]
Check your token rights. It must has been generated with Read+Write permissions minimum.

![image](https://user-images.githubusercontent.com/32335180/154912058-586a558a-b47f-4967-ab1f-1c68c71f1e9b.png)

**Reply by loicdescotte** [reply]
Hi,
In the settings of my app, OAuth is not enable at all : 

![image](https://user-images.githubusercontent.com/638150/154921679-bc1084b9-2b08-4aa1-88b8-8c3521d5cf89.png)

So it seems my code is using basic auth instead, am I right?

Do I need OAuth to enable write permission?
Thanks

**Reply by loicdescotte** [reply]
Thanks, I needed to regenerate the tokens after doing it, and it worked without changing the code
